### PR TITLE
chore: npm dependencies upgrade

### PR DESCRIPTION
**Pull request by Bugs5382**
updated way outdated dependencies for this plugin. fastify (dev), fastify-plugin, and mongodb node driver were way out of date.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
-1

We dont povide package-lock.json with our package. The package.json  says that we need a version bigger or equal 4.0.1 and major is pinned to 4.

So the one who installs @fastify/mongodb can install the latest version.

**Comment by Bugs5382** [comment]
Im confused. I don't see my pushing a package-lock.json file into the code...

**Comment by Uzlopak** [comment]
We disabled on purpose the package-lock generation by .npmrc and also ignore package-lock.json in .gitignore.

We wont accept PRs with package-lock.json anyway.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@fastify/mongodb",
-  "version": "6.2.0",
+  "version": "6.3.0",
```
Comment (jsumners) [inline package.json:3]: Please do not update the package version.

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Fastify MongoDB connection plugin; with this you can share the same MongoDB conn
 
 Under the hood the official [MongoDB](https://github.com/mongodb/node-mongodb-native) driver is used,
 the options that you pass to `register` will be passed to the Mongo client.
-The `mongodb` driver is v4.x.x.
+The `mongodb` driver is [v4.12](https://www.mongodb.com/docs/drivers/node/v4.12/).
```
Comment (jsumners) [inline README.md:11]: Please revert this change.

File: package.json [hunk 3]
```diff
@@ -32,15 +32,15 @@
...
-    "tap": "^16.1.0",
-    "tsd": "^0.24.1"
+    "tap": "^16.3.2",
+    "tsd": "^0.25.0"
   },
   "dependencies": {
-    "fastify-plugin": "^4.0.0",
-    "mongodb": "^4.0.1"
+    "fastify-plugin": "^4.3.0",
+    "mongodb": "^4.12.1"
```
Comment (jsumners) [inline package.json:43]: These changes are unnecessary.

File: package.json [hunk 4]
```diff
@@ -32,15 +32,15 @@
...
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
-    "@types/node": "^18.0.0",
-    "fastify": "^4.0.0-rc.3",
+    "@types/node": "^18.11.9",
+    "fastify": "^4.10.2",
     "standard": "^17.0.0",
-    "tap": "^16.1.0",
-    "tsd": "^0.24.1"
+    "tap": "^16.3.2",
```
Comment (jsumners) [inline package.json:38]: This change is unnecessary.

File: package.json [hunk 5]
```diff
@@ -32,15 +32,15 @@
   "homepage": "https://github.com/fastify/fastify-mongodb#readme",
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
-    "@types/node": "^18.0.0",
-    "fastify": "^4.0.0-rc.3",
+    "@types/node": "^18.11.9",
```
Comment (jsumners) [inline package.json:35]: This change is unnecessary.

**Review by jsumners — CHANGES_REQUESTED** [review]
The updates to `fastify` and `tsd` make sense. Please revert the other changes.

File: package.json (hunk 1 above)
Comment (Bugs5382) [reply package.json:3]: working on it. please hold.

**Comment by Uzlopak** [comment]
@jsumners 
btw. 0.25 tsd was released yesterday. dependabot would update it like always on monday.

File: package.json (hunk 3 above)
Comment (Bugs5382) [reply package.json:43]: mongodb node drive makes perfect and being up to date with the plugin system does. I looked over https://www.mongodb.com/docs/drivers/node/current/compatibility/ and there is not a lot of mongo v3   versions anymore. 4.8.x drive is a compromise because it supports 6. I know that's what I am using in my environment so using 4.0.1 does not.

Comment (Uzlopak) [reply package.json:43]: Again: 
You can install mongo version 4.8.x driver in your project. fastify-mongodb will use than that version. 

All your changes are unnecessary.

Comment (Uzlopak) [reply package.json]: There is no mongodb npm package with version 5.0.0.

With your change, you force all users of @fastify/mongodb to use atleast version 4.8.0 and above.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
This PR forces Implementers to use mongodb native driver to be equal or above mongodb version 4.8.0. 

I disagree with this.

File: package.json [hunk 6]
```diff
@@ -33,14 +33,14 @@
...
     "standard": "^17.0.0",
     "tap": "^16.1.0",
-    "tsd": "^0.24.1"
+    "tsd": "^0.25.0"
   },
   "dependencies": {
-    "fastify-plugin": "^4.0.0",
-    "mongodb": "^4.0.1"
+    "fastify-plugin": "^4.3.0",
+    "mongodb": "^4.8.0"
```
Comment (jsumners) [inline package.json:43]: Please revert these two lines.

**Review by jsumners — CHANGES_REQUESTED** [review]
